marshmallow/marshalling.py

```diff
@@ -42,17 +44,23 @@ def __init__(self):
     def get_errors(self, index=None):
         return self.errors if index is None else self.errors.setdefault(index, {})
 
-    def store_error(self, field_name, error, index=None):
-        self.error_kwargs.update(error.kwargs)
+    def store_error(self, field_name, messages, index=None):
         self.error_field_names.append(field_name)
         errors = self.get_errors(index=index)
         # Warning: Mutation!
-        if isinstance(error.messages, dict):
-            errors[field_name] = error.messages
```

Comment:
> When store_error receives a dict as message, it does not merge it ... This differs from the behaviour of the code duplicated in Unmarshaller. And changing this in store_error would break a lot of field errors tests.

It looks like fields were originally ([8340fca5][0]) allowed to append dicts to the error list, but in a seemingly unrelated bug fix ([4698fe51][1]) it was regressed.

It's probably fine to overwrite some schema errors since field errors have been overwritten for so long, but it might be worth investigating what errors this is hiding. When the field error tests fail, is it because there are extra errors that the tests don't expect, or does it violate core expectations about the structure of the error? Are the hidden errors important?

[0]: https://github.com/marshmallow-code/marshmallow/commit/8340fca5868346e2e20d721d87bbc892354e6d19#diff-c4881f63f8b05b8d3eadf046c3dc7912R110
[1]: https://github.com/marshmallow-code/marshmallow/commit/4698fe51e6d756dca92912a9097b11a7df8a57f2#diff-c4881f63f8b05b8d3eadf046c3dc7912R111